backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ToplevelPhases.kt

```diff
@@ -320,6 +322,20 @@ internal val dependenciesLowerPhase = SameTypeNamedPhaseWrapper(
             }
         })
 
+internal val entryPointPhase = SameTypeNamedPhaseWrapper(
+        name = "addEntryPoint",
+        description = "Add entry point for program",
+        prerequisite = emptySet(),
+        lower = object : CompilerPhase<Context, IrModuleFragment, IrModuleFragment> {
+            override fun invoke(phaseConfig: PhaseConfig, phaserState: PhaserState<IrModuleFragment>,
+                                context: Context, input: IrModuleFragment): IrModuleFragment {
+                assert(context.config.produce == CompilerOutputKind.PROGRAM)
+                context.ir.symbols.entryPoint!!.owner.file.addChild(makeEntryPoint(context))
```

Comment:
Btw, this will not work for entry point in a "cached" dependency. I guess we'll need to generate a synthetic file for this.

Reply:
Yeah, likely so.